src/components/Blogs/BlogDetail.tsx

```diff
@@ -0,0 +1,156 @@
+        <ol className='list-decimal pl-6 mb-4'>{children}</ol>
+      ),
+      [BLOCKS.LIST_ITEM]: (node: Block | Inline, children: React.ReactNode) => (
+        <li className='mb-1'>{children}</li>
+      ),
+      [BLOCKS.EMBEDDED_ASSET]: (node: Block | Inline) => {
+        const assetId = node.data.target.sys.id;
+        const asset = blog.description.links.assets.block.find(
+          (asset) => asset.sys.id === assetId
+        );
+
+        if (!asset) {
+          return null;
+        }
+
+        return (
+          <div className='flex justify-center items-center my-4'>
+            <Zoom classDialog='custom-zoom' zoomMargin={40}>
+              <img
+                alt={asset.title || 'Blog image'}
+                src={asset.url}
+                width={asset.width || 500}
+                height={asset.height || 300}
+                className='object-cover relative w-[200px] h-[200px] transition-all ease-in duration-150 rounded-2xl'
+              />
+            </Zoom>
+          </div>
+        );
+      },
+      [INLINES.HYPERLINK]: (
+        node: Block | Inline,
+        children: React.ReactNode
+      ) => {
+        return (
+          <a
+            href={node.data.uri}
+            className='text-blue-600 hover:underline'
+            target='_blank'
+            rel='noopener noreferrer'
+          >
+            {children}
+          </a>
+        );
+      },
+      [INLINES.EMBEDDED_ENTRY]: (
+        node: Block | Inline,
+        children: React.ReactNode
+      ) => {
+        // Handle embedded entries here if needed
+        return <span>{children}</span>;
+      },
+      [BLOCKS.QUOTE]: (node: Block | Inline, children: React.ReactNode) => {
+        // Sometimes code blocks are converted to blockquotes during markdown conversion
+        return (
+          <blockquote className='border-l-4 border-gray-300 pl-4 italic my-4'>
+            {children}
+          </blockquote>
+        );
+      },
+    },
+    renderMark: {
+      [MARKS.CODE]: (text) => {
+        // This handles inline code (`code`)
+        if (typeof text === 'string' && text.includes('\n')) {
+          const language = 'javascript';
+          return (
+            <div className='my-4  group relative z-1'>
+              <SyntaxHighlighter
+                style={agate}
+                language={language}
+                customStyle={{ borderRadius: '12px' }}
+                PreTag='div'
+              >
+                {text}
+              </SyntaxHighlighter>
+              <div className='absolute z-[100] top-2 right-2'>
+                <CopyButton text={text} notVisible />
+              </div>
+            </div>
+          );
+        }
+        return (
+          <code className='bg-gray-200 dark:bg-gray-700 text-foreground dark:text-primary font-mono p-1 rounded-md'>
+            {text}
+          </code>
+        );
+      },
+    },
+  };
```

Comment:
The `react-syntax-highlighter` library is a great choice for code blocks. However, it seems to be missing a language detection feature, which could improve the highlighting for different code snippets. Adding a language detection library or manually specifying the language for each block could be a good enhancement.